### PR TITLE
Depthmap toolkit - Visualisation refactor

### PR DESCRIPTION
**Pull request by lvonasek**
# Description

This refactor enables to use the visualisations in other scripts. E.g. to get normal map call:
```
    output = np.zeros((dmap.width, dmap.height, 3))
    render_normal(output, 0, dmap)
```

### CODE REVIEW

File: src/common/depthmap_toolkit/tests/test_visualisation.py [hunk 1]
```diff
@@ -23,32 +23,29 @@ def test_blur_face():
...
-    output_blurred = blur_face(output_unblurred, highest, dmap)
+    output_blurred = blur_face(output_unblurred, 0, highest, dmap)
 
     # Assert some pixels in whole image change (image not same)
-    all_count = dmap.width * dmap.height * SUBPLOT_COUNT
+    all_count = dmap.width * dmap.height
     count = np.count_nonzero(output_unblurred - output_blurred) / 3
     ratio_blurred = count / all_count
-    assert 0.003 < ratio_blurred < 0.9
+    assert 0.01 < ratio_blurred < 0.9
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/tests/test_visualisation.py:37]: Why was this changed?

File: src/common/depthmap_toolkit/tests/test_visualisation.py [hunk 2]
```diff
@@ -23,32 +23,29 @@ def test_blur_face():
...
-    object_y = offset + int(dmap.height * (1.0 - OFFSET_X_Y[1]))
+    object_y = int(dmap.height * (1.0 - OFFSET_X_Y[1]))
     slice_x = slice(object_x - 2, object_x + 2)
     slice_y = slice(object_y - 2, object_y + 2)
     assert (output_unblurred[slice_x, slice_y] != output_blurred[slice_x, slice_y]).any()
 
     # Assert that NOT blurred around corner
     corner_x = 0
-    corner_y = offset
+    corner_y = 0
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/tests/test_visualisation.py:48]: Great, this makes the code easier!

File: src/common/depthmap_toolkit/visualisation.py [hunk 3]
```diff
@@ -52,78 +52,102 @@ def blur_face(data: np.array, highest: list, dmap: Depthmap) -> np.array:
...
 
-def render_pixel(output: np.array,
-                 x: int,
-                 y: int,
-                 floor: float,
-                 mask: np.array,
+def render_confidence(output: np.array,
+                      subplot: int,
+                      dmap: Depthmap):
+
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/visualisation.py:67]:
```suggestion
```

File: src/common/depthmap_toolkit/visualisation.py [hunk 4]
```diff
@@ -52,78 +52,102 @@ def blur_face(data: np.array, highest: list, dmap: Depthmap) -> np.array:
...
+        for y in range(dmap.height):
+            index = subplot * dmap.height + dmap.height - y - 1
+            output[x][index][:] = dmap.parse_confidence(x, y)
+            if output[x][index][0] == 0:
+                output[x][index][:] = 1
+
+
+def render_depth(output: np.array,
+                 subplot: int,
                  dmap: Depthmap):
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/visualisation.py:79]:
```suggestion
```

File: src/common/depthmap_toolkit/visualisation.py [hunk 5]
```diff
@@ -52,78 +52,102 @@ def blur_face(data: np.array, highest: list, dmap: Depthmap) -> np.array:
...
+            if not depth:
+                continue
+            index = subplot * dmap.height + dmap.height - y - 1
+            output[x][index] = min(max(0, 1.0 - min(depth / 2.0, 1.0)), 1)
+
+
+def render_normal(output: np.array,
+                  subplot: int,
+                  dmap: Depthmap):
+
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/visualisation.py:92]:
```suggestion
```

File: src/common/depthmap_toolkit/visualisation.py [hunk 6]
```diff
@@ -52,78 +52,102 @@ def blur_face(data: np.array, highest: list, dmap: Depthmap) -> np.array:
...
+            index = subplot * dmap.height + dmap.height - y - 1
+            output[x][index][0] = abs(normal[0])
+            output[x][index][1] = abs(normal[1])
+            output[x][index][2] = abs(normal[2])
+
+
+def render_rgb(output: np.array,
+               subplot: int,
+               dmap: Depthmap):
+
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/visualisation.py:105]:
```suggestion
```

File: src/common/depthmap_toolkit/visualisation.py [hunk 7]
```diff
@@ -52,78 +52,102 @@ def blur_face(data: np.array, highest: list, dmap: Depthmap) -> np.array:
...
+            output[x][index][1] = dmap.rgb_array[y][x][1] / 255.0
+            output[x][index][2] = dmap.rgb_array[y][x][2] / 255.0
+
+
+def render_segmentation(output: np.array,
+                        subplot: int,
+                        floor: float,
+                        mask: np.array,
+                        dmap: Depthmap):
+
```
Comment (markus-hinsche) [inline src/common/depthmap_toolkit/visualisation.py:119]:
```suggestion
```

**Review by markus-hinsche — APPROVED** [review]
LGTM

I just suggested minor code style improvements

File: src/common/depthmap_toolkit/tests/test_visualisation.py (hunk 1 above)
Comment (lvonasek) [reply src/common/depthmap_toolkit/tests/test_visualisation.py:37]: there are not 5 subplots used anymore...
